server/data/data-postgres/src/main/java/org/apache/james/domainlist/postgres/PostgresDomainList.java

```diff
@@ -34,16 +34,24 @@
 import org.apache.james.domainlist.lib.AbstractDomainList;
 import org.jooq.exception.DataAccessException;
 
+import com.google.common.annotations.VisibleForTesting;
+
 import reactor.core.publisher.Flux;
 import reactor.core.publisher.Mono;
 
 public class PostgresDomainList extends AbstractDomainList {
     private final PostgresExecutor postgresExecutor;
 
     @Inject
-    public PostgresDomainList(DNSService dnsService, JamesPostgresConnectionFactory postgresConnectionFactory) {
+    public PostgresDomainList(DNSService dnsService, PostgresExecutor.Factory postgresExecutorFactory) {
+        super(dnsService);
+        this.postgresExecutor = postgresExecutorFactory.create(Optional.empty());
```

Comment:
We will do this in every class? How many connections will we open?

I am not fan

Maybe play with @Provides in guice modules and use `@Named("initialized") @Singleton` in order to denote the fact the the postgresExecutor is initialized?

IE

```
@Provides
@Named("initialized")
@Singleton
PostgresExecutor createTables(PostgresExecutor executor, PostgresTableManager postgresTableManager) {
    // Create tables
}


Then

@Inject 
 public PostgresDomainList(DNSService dnsService, @Named("Initialized") PostgresExecutor postgresExecutor) {
    // ...
 }
 ```

Reply:
> How many connections will we open?

On that, actually, it is still the same connection. Our connection factory makes sure that it is the same connection even called multiple times concurrently.

Reply:
As Quan said we return the same connection... Will check the way you propose though @chibenwa

Reply:
@chibenwa if you don't mind... is it ok to create a ticket for this? I tried honestly and I had an other binding issue. Tried to tinker it in different ways and had other issues... Seems a bit trickier IMO and I'm being distracted too much lately with other stuff to really put my mind into it

Reply:
Nvm the code changed again with Postgresexecutor, need to rework it...

Reply:
Does not work now... With latest code on PostgresExecutor Factory, because the `@Named(DEFAULT_INJECT) PostgresExecutor defaultPostgresExecutor` is being injected in PostgresTableManager, same PostgresExecutor that is used for PostgresDomainList... 

If you do this above now you have a circle dependency on POstgresexecutor, so Guice complains

Reply:
Let's refactor this later (in another ticket) IMO.

Reply:
Nope cause it fails like this, back to previous issue, where domain is trying to get conf done before tables get initialized. The trick you did last week does not work anymore because of the recent refactor on postgresExecutor. Might maybe need to do something like with cassandra